src/EchoToImageLUT.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef RADAR_BASE_ECHO_TO_IMAGE_LUT_HPP
+#define RADAR_BASE_ECHO_TO_IMAGE_LUT_HPP
+
+#include <opencv2/core.hpp>
+#include <vector>
+
+namespace radar_base {
+    class EchoToImageLUT {
+        /** m_data[angle * sweepSize + cellIndex] contains the list of points in the image
+         * that should be filled by the echo with the given angle and cell
+         */
+        std::vector<cv::Point> m_data;
+        std::vector<int> m_data_index;
+
+        int m_num_angles;
+        int m_sweep_size;
+
+        static int normalizeAngle(int angle, int num_angles);
+
+        typedef std::vector<std::vector<cv::Point>> RawTable;
+        static RawTable computeRawLUTTable(int sweep_size,
+            int window_size,
+            int num_angles,
+            float beam_width);
+        static void addRawLUTEntry(RawTable& table,
+            int angle,
+            int echo_index,
+            int sweep_size,
+            int num_angles,
+            cv::Point2i const& p);
+
+        void setup(int sweep_size, int window_size, float beam_width);
+
+    public:
+        EchoToImageLUT() = delete;
+
+        EchoToImageLUT(int num_angles, int sweep_size, float beam_width, int window_size);
```

Comment:
Add some documentation explaining the parameters. `num_angles` and `beam_width` might seem weird if some radar doesnt return data like Furuno, so make sure it's clear.